lib/fluent/plugin/in_tail.rb

```diff
@@ -303,7 +308,31 @@ def expand_paths
                end
         path.include?('*') ? Dir.glob(path) : path
       }.flatten.uniq
-      paths.uniq - excluded
+      # filter out non existing files, so in case pattern is without '*' we don't do unnecessary work
+      hash = {}
+      (paths - excluded).select { |path|
+        FileTest.exist?(path)
+      }.each { |path|
+        tuple = PathInodeTuple.new(path, Fluent::FileWrapper.stat(path).ino)
+        if @follow_inodes
+          hash[tuple.ino] = tuple
+        else
+          hash[tuple.path] = tuple
+        end
+      }
+      hash
+    end
+
+    def existence_path
+      hash = {}
+      @tails.keys.each {|path_ino|
```

Comment:
How about `each_key` ?

Reply:
Fixed in https://github.com/fluent/fluentd/pull/3182/commits/d8019edd6c3625bd7e0fda7f5109becaa4909ac1.